### ISSUE TITLE
`ext` not re-usable in profile.

### ISSUE THREAD

**Issue by fosdev** [alps-io/spec#7]
Currently, the `ext` tag needs an href or something to allow it to be re-usable. Further, I think it needs to support a more abstract meaning potentially than something as concrete as the example here: https://github.com/alps-io/spec/blob/master/draft-00/draft-00.txt#L978

It may already do that, but I think an example would be good, where one may want to indicate some metadata about a list of options, vs. the actual options which may change at runtime.

Also,  seems like some more instructions as to what should be an an extension document as an example. If it is just human readable name, why the need for the human readable tag in extension itself.

**Comment by fosrias** [comment]
@mamund Couple of thoughts.

1. I should be able to describe an extension in ALPS document once (stay DRY) and re-use in in multiple places in that same document.  This is tantamount to defining a top-level descriptor and then re-using it throughout the document as a nested descriptor, for example.

An example is if I am using an extension to say that a descriptor is a list data type or an enum, I may want to decorate more than one descriptor that way in a document.

2. If can re-use an `ext` element defined in an ALPS document, I SHOULD be able to reference an `ext` element defined in another ALPS document (maybe an extension only ALPS document).

The issue here is this, re-use, that is referencing is accomplished by `href` to the `id` of another element generally in ALPS. This works fine for a descriptor, but not `ext` because `href` is reserved for documentation of the extension.

What is being extended, an `alps` element or a `description` element and as such I think it should reflect how those are documented, i.e. use a `doc` element and reserve `href` for referencing.

I think the right way to solve all this is something like this:
```
The 'ext' element has the following properties.

1.  'id'

2.  'href'

3.  'value'

4.   'doc'

The 'id' property is REQUIRED.  The 'href' is OPTIONAL.  
The 'value' property is OPTIONAL.  The 'doc' is RECOMMENDED
since the content is undetermined; its meaning and use SHOULD 
be explained by a `doc`.

If present, the 'href' property MUST be a dereferenceable URL, that
points to another 'ext' either within the current ALPS
document or in another ALPS document.
```
This would be a breaking change in the current spec, but I think it is the correct way to do this.

**Comment by mamund** [comment]
Sounds reasonable. Write up some examples to show how it works and we can discuss.

**Comment by fosrias** [comment]
Couple possibilities:

```xml
<alps>
  <ext id="reusable-enum-def" value="on off" />
  <descriptor id="switch" type="semantic">
    <ext href="#reusable-enum-def/>
  </descriptor>
  <descriptor id="light-bulb" type="semantic">
    <ext href="#reusable-enum-def"/>
  </descriptor>
</alps>
```
or support `ext` as `xml` property

```xml
<alps>
  <ext id="reusable-enum-def" value="on off" />
  <descriptor id="switch" type="semantic" ext="#reusable-enum-def"/>
  <descriptor id="light-bulb" type="semantic" ext="#reusable-enum-def"/>
</alps>
```

**Comment by fosrias** [comment]
@mamund These examples clarify?

**Comment by mamund** [comment]
not yet<g>. 

first one make sense but second one (as attribute) doesn't yet. i'm wary of using EXT references as they are not returning the same thing as other refs (DOC and LINK). we may be running into challenges on what to expect in reply 

there's a REL on LINK, there's an implied expectation on DOC and now w/ EXT as a possible href there are additional challenges.

we should proly discuss this one live sometime to sort things out, eh?

**Comment by fosrias** [comment]
Yes. We are not consistent on uses of the same thing @mamund across different components
